File: src/main/kotlin/io/mcm/springbootwithkotlin/model/entities/Book.java
package io.mcm.springbootwithkotlin.model.entities;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name = "books")
public class Book {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    Long id;
    Double price;
    Date publishedDate;
    String author;
    Boolean isIssued;

    @Override
    public String toString() {
        return "Book{" + "id=" + id + ", price=" + price + ", publishedDate=" + publishedDate + ", author='" + author + '\'' + ", issued=" + isIssued + '}';
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Boolean getIsIssued() {
        return isIssued;
    }

    public void setIsIssued(Boolean isIssued) {
        this.isIssued = isIssued;
    }

    public Double getPrice() {
        return price;
    }

    public void setPrice(Double price) {
        this.price = price;
    }

    public Date getPublishedDate() {
        return publishedDate;
    }

    public void setPublishedDate(Date publishedDate) {
        this.publishedDate = publishedDate;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }
}

File: src/main/kotlin/io/mcm/springbootwithkotlin/model/entities/Book.kt
package io.mcm.springbootwithkotlin.model.entities

import java.util.*
import javax.persistence.*

@Entity
@Table(name = "books")
open class Book {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    var id: Long? = null
    var price: Double? = null
    var publishedDate: Date? = null
    var author: String? = null
    var isIssued: Boolean? = null

    constructor()
    constructor(
        id: Long? = null,
        price: Double? = null,
        publishedDate: Date? = null,
        author: String? = null,
        isIssued: Boolean? = null
    ) {
        this.id = id
        this.price = price
        this.publishedDate = publishedDate
        this.author = author
        this.isIssued = isIssued
    }

    override fun toString(): String {
        return "Book{id=$id, price=$price, publishedDate=$publishedDate, author='$author', issued=$isIssued}"
    }
}